clients/src/main/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumer.java

```diff
@@ -707,6 +708,8 @@ public ConsumerRecords<K, V> poll(final Duration timeout) {
                 updateAssignmentMetadataIfNeeded(timer);
                 final Fetch<K, V> fetch = pollForFetches(timer);
                 if (!fetch.isEmpty()) {
+                    sendFetches(timer);
```

Comment:
at this point we may already have records in hand to return (consumed position updated), so we should be very careful to not throw any error here. But this `sendFetches` could throw interrupted because of the `addAndGet` right? 

Shouldn't we just do a best effort to pipeline the next requests using `add` instead of `addAndGet`? It would achieve what we want, removing the risk of errors, and it would actually align better with what the classic does on this `sendFetches` + `transmitSends`:
https://github.com/apache/kafka/blob/140d35c5459f4c7a91b23a238467cb5aa01b59fb/clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerNetworkClient.java#L327-L329

Reply:
Done.